File: backend/src/main/java/com/hardware/erp/project/service/WorkTypeService.java
package com.hardware.erp.project.service;

import com.hardware.erp.project.dto.WorkTypeRequest;
import com.hardware.erp.project.dto.WorkTypeResponse;

import java.util.List;

public interface WorkTypeService {

    List<WorkTypeResponse> list();

    WorkTypeResponse create(WorkTypeRequest request);

    WorkTypeResponse update(Long id, WorkTypeRequest request);
}

File: frontend/src/modules/project/services/workTypeService.ts
import { apiGet, apiPost, apiPut } from '@/services/apiClient';
import type { WorkTypeRequest, WorkTypeResponse } from '../types';

/** Backend: project/controller/WorkTypeController.java */
export const workTypeService = {
  list: () => apiGet<WorkTypeResponse[]>('/v1/work-types'),

  create: (body: WorkTypeRequest) => apiPost<WorkTypeResponse>('/v1/work-types', body),

  update: (id: number, body: WorkTypeRequest) => apiPut<WorkTypeResponse>(`/v1/work-types/${id}`, body),
};
